src/core/umount.c

```diff
@@ -415,14 +452,13 @@ static int mount_points_list_umount(MountPoint **head, bool *changed, bool log_e
                          * explicitly remount the super block of that
                          * alias read-only we hence should be
                          * relatively safe regarding keeping dirty an fs
-                         * we cannot otherwise see. */
-                        log_info("Remounting '%s' read-only with options '%s'.", m->path, options);
-                        if (mount(NULL, m->path, NULL, MS_REMOUNT|MS_RDONLY, options) < 0) {
-                                if (log_error)
-                                        log_notice_errno(errno, "Failed to remount '%s' read-only: %m", m->path);
-                                if (nonunmountable_path(m->path))
-                                        n_failed++;
-                        }
+                         * we cannot otherwise see.
+                         *
+                         * Since the remount can hang in the instance of
+                         * remote filesystems, we remount asynchronously
+                         * and skip the subsequent umount if it fails */
+                        if (remount_with_timeout(m, DEFAULT_TIMEOUT_USEC, options, &n_failed, log_error) < 0)
+                                continue;
```

Comment:
hmm, the nonunmountable_path() thing got lost here?

Reply:
Yep, I figured it was redundant based on the immediate second nonunmountable_path check outside of the container verification. Looking at it now, I can see why we might want to include it for an accurate n_failed count on return. Adding it back in with a subsequent push.